doc/fundamentals/creating_analysis_pipelines.md

```diff
@@ -0,0 +1,62 @@
+
+For the reading files part, Pachyderm automatically mounts each input data repository as `/pfs/<repo_name>` in the running instances of your Docker image (called "containers"). The code that you write just needs to read input data from this directory, just like in any other file system.  Your analysis code also does NOT have to deal with data sharding or parallelization as Pachyderm will automatically shard the input data across parallel containers. For example, if you've got four containers running your Python code, Pachyderm will automatically supply 1/4 of the input data to `/pfs/<repo_name>` in each running container. That being said, you also have a lot of control over how that input data is split across containers. Check out our guide on :doc: `parallelization` to see the details of that.
+
+For the writing files part (saving results, etc.), you code simply needs to write to `/pfs/out`. This is a special directory mounted by Pachyderm in all of your running containers. Similar to reading data, your code doesn't have to manage parallelization or sharding, just write data to `/pfs/out` and Pachyderm will make sure it all ends up in the correct place. 
+
+## 2. Building a Docker Image
+
+When you create a Pachyderm pipeline (which will be discussed next), you need to specify a Docker image including the code or binary you want to run.  Please refer to the [official documentation](https://docs.docker.com/engine/tutorials/dockerimages/) to learn how to build a Docker images.  Note, your Docker image should NOT specifiy a `CMD`.  Rather, you specify what commands are to be run in the container when you create your pipeline.
+
+Unless Pachyderm is running on the same host that you used to build your image, you'll need to use a public or private registry to get your image into the Pachyderm cluster.  Once (free) option is to use Docker's DockerHub registry.  You can refer to the [official documentation](https://docs.docker.com/engine/tutorials/dockerimages/#/push-an-image-to-docker-hub) to learn how to push your images to DockerHub. That being said, you are more than welcome to use any other public or private Docker registry.
+
+## 3. Creating a Pipeline
+
+Now that you've got your code and image built, the final step is to tell Pachyderm to run the code in your image on certain input data.  To do this, you need to supply Pachyderm with a JSON pipeline specification. There are four main components to a pipeline specification: name, transform, parallelism and inputs. Detailed explanations of the specification parameters and how they work can be found in the [pipeline specification docs](../reference/pipeline_spec). 
+
+Here's an example pipeline spec:
+```json
+{
+  "pipeline": {
+    "name": "wordcount"
+  },
+  "transform": {
+    "image": "wordcount-image",
+    "cmd": ["/binary", "/pfs/data", "/pfs/out"]
```

Comment:
I think this is just meant to be a generic way of saying you can specify a `binary` that runs on data in PFS and writes out to PFS at the particular places.